apps/whale-api/src/module.api/poolpair.service.ts

```diff
@@ -70,32 +72,31 @@ export class PoolPairService {
   }
 
   /**
-   * TODO(fuxingloh): graph based matrix resolution
-   * Currently implemented with fix pair derivation
-   * Ideally should use vertex directed graph where we can always find total liquidity if it can be resolved.
+   * Graph based matrix resolution
+   * Returns the total liquidity of the poolpair by finding its best path to USDT
    */
   async getTotalLiquidityUsd (info: PoolPairInfo): Promise<BigNumber | undefined> {
-    const [a, b] = info.symbol.split('-')
-    if (['DUSD', 'USDT', 'USDC'].includes(a)) {
-      return info.reserveA.multipliedBy(2)
-    }
+    const usdtToken = await this.deFiDCache.getTokenInfoBySymbol('USDT')
 
-    if (['DUSD', 'USDT', 'USDC'].includes(b)) {
-      return info.reserveB.multipliedBy(2)
+    if (usdtToken === undefined) {
+      throw new NotFoundException('Unable to find USDT token')
```

Comment:
Should we throw error here? Or just keep it as undefined like the current behavior